modules/gdscript/gdscript_parser.cpp

```diff
@@ -4358,8 +4358,6 @@ Error GDScriptParser::_parse(const String &p_base_path) {
 
 	base_path = p_base_path;
 
-	clear();
```

Comment:
While so far I didn't run into any issues with this, I'm not sure why clear() was added here in the first place, so I don't know if that change is ok.
If not, maybe moving the validating = false and for_completion = false lines from clear() into the constructor is a better solution?

Reply:
That `clear()` is strange indeed. Another solution might be to add that `clear()` to the beginning of `parse()` and `parse_bytecode()`. Then it's still called on each parse, but it shouldn't overwrite anything.

Reply:
Having a closer look at `parse()` and `parse_bytecode()` it looks like they duplicate a lot of what's happening inside clear already ... so when moving `clear()` to their beginning, then those duplicates could be cleaned up as well.

Reply:
Sounds good to me.